src/rule-must-colocate-fragment-spreads.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+'use strict';
+
+const {visit} = require('graphql');
+const utils = require('./utils');
+
+function getGraphQLFragmentNames(graphQLAst) {
+  const fragmentNames = {};
+  visit(graphQLAst, {
+    FragmentSpread(node, key, parent, path, ancestors) {
+      for (const ancestorNode of ancestors) {
+        if (ancestorNode.kind === 'OperationDefinition') {
+          if (
+            ancestorNode.operation === 'mutation' ||
+            ancestorNode.operation === 'subscription'
+          ) {
+            return;
+          }
+        }
+      }
+      for (const directiveNode of node.directives) {
+        if (directiveNode.name.value === 'module') {
+          return;
+        }
+        if (directiveNode.name.value === 'relay') {
+          for (const argumentNode of directiveNode.arguments) {
+            if (
+              argumentNode.name.value === 'mask' &&
+              argumentNode.value.value === false
+            ) {
+              return;
+            }
+          }
+        }
+      }
+      const nameNode = node.name;
```

Comment:
You can store the `FragmentSpread` node itself, instead of its `Name` node. This makes the error reporting more accurate, and in the future if we add a fixer, it will remove the entire fragment spread (including directives), not just the name.